poly/src/polynomial/univariate/sparse.rs

```diff
@@ -62,11 +62,29 @@ impl<F: Field> Polynomial<F> for SparsePolynomial<F> {
         if self.is_zero() {
             return F::zero();
         }
+
+        // Add one for good luck/avoid corner case
```

Comment:
Why the `+ 1`? What's the corner case here?

Reply:
The corner case is when degree is a power of two.

Reply:
It's not important to work it out since the efficiency difference is logarithmic, so it only barely matters even when nonzero terms are logarithmic in the degree.

Reply:
Ok, I've used the exact formula now but better written

Reply:
So you want to calculate `log2_floor`? (As opposed to `log2_ceil`, which is what `ark_std::log2` implements)

Reply:
Well, no, it's actually log2_plus_one_floor

Reply:
Er how so? This playground shows that it's just `log2_ceil`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=90a1770db078df16538f7e1a304524d7

Reply:
Oh nvm, I get it now.

Reply:
Hmm then I'm confused; why not use `ark_std::log2`? Why do you need to round up 2^{k} to 2^{k + 1}?

Reply:
I see, it's because of the loop which goes up to `max_pows - 1`